Framework/Crystal/src/PredictSatellitePeaks.cpp

```diff
@@ -319,9 +329,17 @@ void PredictSatellitePeaks::predictOffsets(int indexModulatedVector, V3D offsets
       continue;
 
     IPeak_uptr iPeak = Peaks->createPeak(Qs, 1);
+
+    iPeak->setGoniometerMatrix(goniometer);
     Peak_uptr peak(static_cast<DataObjects::Peak *>(iPeak.release()));
 
-    peak->setGoniometerMatrix(goniometer);
+    if( peak == nullptr )
+    {
+      std::cout << "Ahoy! position 0 LeanElasticPeak triggered the condition" << std::endl;
+      continue;
+    }
+    else { std::cout << "Ahoy position 0 Peak did not trigger the condition, checking detectors..." 
+           << std::endl; }
 
     if (!peak->findDetector(tracer))
```

Comment:
`LeanElasticPeak` doesn't have detector information so this should always fail for that type.